teamscale-client/src/main/java/com/teamscale/client/HttpUtils.java

```diff
@@ -66,13 +75,68 @@ public static Retrofit createRetrofit(Consumer<Retrofit.Builder> retrofitBuilder
 		OkHttpClient.Builder httpClientBuilder = new OkHttpClient.Builder();
 		setTimeouts(httpClientBuilder, readTimeout, writeTimeout);
 		setUpSslValidation(httpClientBuilder);
+		setUpProxyServer(httpClientBuilder);
 		okHttpBuilderAction.accept(httpClientBuilder);
 
 		Retrofit.Builder builder = new Retrofit.Builder().client(httpClientBuilder.build());
 		retrofitBuilderAction.accept(builder);
 		return builder.build();
 	}
 
+	/**
+	 * Java and/or OkHttp do not pick up the http.proxy* and https.proxy* system properties reliably. We need to teach
+	 * OkHttp to always pick them up.
+	 * <p>
+	 * Sources: <a
+	 * href="https://memorynotfound.com/configure-http-proxy-settings-java/">https://memorynotfound.com/configure-http-proxy-settings-java/</a>
+	 * &
+	 * <a href="https://stackoverflow.com/a/35567936">https://stackoverflow.com/a/35567936</a>
+	 */
+	private static void setUpProxyServer(OkHttpClient.Builder httpClientBuilder) {
+		boolean setHttpsProxyWasSuccessful = setUpProxyServerForProtocol(ProxySystemProperties.Protocol.HTTPS,
+				httpClientBuilder);
+		if (!setHttpsProxyWasSuccessful) {
+			setUpProxyServerForProtocol(ProxySystemProperties.Protocol.HTTP, httpClientBuilder);
+		}
+	}
+
+	private static boolean setUpProxyServerForProtocol(ProxySystemProperties.Protocol protocol,
+													   OkHttpClient.Builder httpClientBuilder) {
+
+		ProxySystemProperties proxySystemProperties = new ProxySystemProperties(protocol);
+		String proxyHost = proxySystemProperties.getProxyHost();
+		int proxyPort = proxySystemProperties.getProxyPort();
+		String proxyUser = proxySystemProperties.getProxyUser();
+		String proxyPassword = proxySystemProperties.getProxyPassword();
+
+		if (proxySystemProperties.proxyServerIsSet()) {
+			setProxyServer(httpClientBuilder, proxyHost, proxyPort);
+
+			if (proxySystemProperties.proxyAuthIsSet()) {
+				setProxyAuthenticator(httpClientBuilder, proxyUser, proxyPassword);
+			}
+
+			return true;
+		}
+		return false;
+
+	}
+
+	private static void setProxyServer(OkHttpClient.Builder httpClientBuilder, String proxyHost, int proxyPort) {
+		httpClientBuilder.proxy(new Proxy(Proxy.Type.HTTP, new InetSocketAddress(proxyHost, proxyPort)));
```

Comment:
(very) nitpicky: I was a little taken confused because "proxyServerIsSet" is true and then we want to "setProxyServer" so for a short moment I thought if we need to inverse the condition. Maybe change the name to 
```suggestion
	private static void useProxyServer(OkHttpClient.Builder httpClientBuilder, String proxyHost, int proxyPort) {
		httpClientBuilder.proxy(new Proxy(Proxy.Type.HTTP, new InetSocketAddress(proxyHost, proxyPort)));
```